subsys/sensing/sensor/phy_3d_sensor/phy_3d_sensor.c

```diff
@@ -87,14 +245,17 @@ static const struct sensing_sensor_register_info phy_3d_sensor_reg = {
 };
 
 #define SENSING_PHY_3D_SENSOR_DT_DEFINE(_inst)				\
-	static struct phy_3d_sensor_context _CONCAT(ctx, _inst) = {	\
+	static struct phy_3d_sensor_data _CONCAT(data, _inst);		\
+	static const struct phy_3d_sensor_config _CONCAT(cfg, _inst) = {\
 		.hw_dev = DEVICE_DT_GET(				\
 				DT_PHANDLE(DT_DRV_INST(_inst),		\
 				underlying_device)),			\
 		.sensor_type = DT_PROP(DT_DRV_INST(_inst), sensor_type),\
 	};								\
-	SENSING_SENSOR_DT_DEFINE(DT_DRV_INST(_inst),			\
-		&phy_3d_sensor_reg, &_CONCAT(ctx, _inst),		\
+	SENSING_SENSOR_DT_INST_DEFINE(_inst, &phy_3d_sensor_reg, NULL,	\
+		&phy_3d_sensor_init, NULL,				\
+		&_CONCAT(data, _inst), &_CONCAT(cfg, _inst),		\
+		POST_KERNEL, CONFIG_SENSOR_INIT_PRIORITY,		\
```

Comment:
This is a critical nuance, the physical sensors should for example when using the ICM42688 create:

1. sensor device (`const struct device`)
2. 3 separate `const struct sensing_sensor_info` objects for `ACCEL`, `GYRO`, and `DIE_TEMPERATURE`.

The current model creates a separate `device` for each data type

Reply:
Yes, I'll fix it.

I would like to do the following change:
1. Change the type of `sensor-type` from `int` to `array`
```diff
# Copyright (c) 2023, Intel Corporation
# SPDX-License-Identifier: Apache-2.0
#

description: Sensing subsystem sensor common properties bindings.

include: sensor-device.yaml

properties:
---  sensor-type:
+++  sensor-types:
---    type: int
+++    type: array
    required: true
    description: sensor type id (follow HID spec definition)
```
2. In `SENSING_SENSOR_DEFINE`, it will create multiple instances of `sensing_sensor` by enumerating `sensor-types`.

Reply:
I think there's a key distinction here, you need 1 sensor but multiple sensor_info nodes. Please see my sample PR at https://github.com/zephyrproject-rtos/zephyr/pull/56963

Reply:
I updated the PR, and add the commit [sensing: support multiple sensor types in one device](https://github.com/zephyrproject-rtos/zephyr/pull/64478/commits/b4aeea192af59bcbec2ef50e2a0ebe769fb7e551)
In this commit, the single bmi160 device would create two sensing_sensor instances for `Accel` and `Gyro`.